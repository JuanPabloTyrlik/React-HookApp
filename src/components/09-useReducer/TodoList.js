import React from 'react';
import PropTypes from 'prop-types';
import TodoListItem from './TodoListItem';

const TodoList = ({ todoList, handleDelete, handleToggle }) => {
    return (
        <ul className="list-group list-group-flush">
            {todoList.map((todo, i) => (
                <TodoListItem
                    key={todo.id}
                    todo={todo}
                    i={i}
                    handleToggle={handleToggle}
                    handleDelete={handleDelete}
                />
            ))}
        </ul>
    );
};

TodoList.propTypes = {
    todoList: PropTypes.array.isRequired,
    handleDelete: PropTypes.func.isRequired,
    handleToggle: PropTypes.func.isRequired,
};

export default TodoList;
