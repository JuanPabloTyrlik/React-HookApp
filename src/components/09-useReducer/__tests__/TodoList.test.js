import { shallow } from 'enzyme';
import React from 'react';
import TodoList from '../TodoList';

describe('Tests on TodoList', () => {
    const demoTodos = [
        {
            id: 1,
            desc: 'Aprender React',
            done: true,
        },
        {
            id: 2,
            desc: 'Aprender Mongo',
            done: false,
        },
    ];
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();
    const wrapper = shallow(
        <TodoList
            todoList={demoTodos}
            handleDelete={handleDelete}
            handleToggle={handleToggle}
        />
    );
    test('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
        expect(wrapper.find('TodoListItem').at(0).prop('todo')).toEqual(
            demoTodos[0]
        );
        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(
            expect.any(Function)
        );
        expect(wrapper.find('TodoListItem').at(0).prop('handleToggle')).toEqual(
            expect.any(Function)
        );
    });
});
