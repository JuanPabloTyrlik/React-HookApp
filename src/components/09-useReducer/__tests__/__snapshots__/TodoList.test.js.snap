// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on TodoList should render correctly 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender React",
        "done": true,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
